audit-service/src/main/java/uk/gov/gchq/palisade/service/audit/stream/SerDesConfig.java

```diff
@@ -82,14 +93,28 @@ private SerDesConfig() {
     /**
      * Kafka value deserialiser for upstream messages coming in as input
      *
+     * @param configProperties contains the directory for error files
      * @return an appropriate value deserialiser for the topic's message content (AuditErrorMessage)
      */
-    public static Deserializer<AuditErrorMessage> errorValueDeserializer() {
+    public static Deserializer<AuditErrorMessage> errorValueDeserializer(final AuditServiceConfigProperties configProperties) {
         return (String ignored, byte[] auditRequest) -> {
             try {
                 return MAPPER.readValue(auditRequest, AuditErrorMessage.class);
             } catch (IOException e) {
-                throw new SerializationFailedException(DESERIALIZATION_FAILED_MESSAGE + new String(auditRequest, Charset.defaultCharset()), e);
+                String failedAuditString = new String(auditRequest, Charset.defaultCharset());
+                String fileName = "Error-" + ZonedDateTime.now(ZoneOffset.UTC)
+                        .format(DateTimeFormatter.ISO_INSTANT).replace(":", "-");
+                File directory = new File(configProperties.getErrorDirectory());
+                File parent = directory.getAbsoluteFile().getParentFile();
+                File timestampedFile = new File(parent, fileName);
+                try (FileWriter fileWriter = new FileWriter(timestampedFile, Charset.defaultCharset(), !timestampedFile.createNewFile())) {
+                    fileWriter.write(failedAuditString);
+                    LOGGER.info("Failed to deserialize an 'error' audit message. Created file {}", timestampedFile);
+                } catch (IOException ioException) {
+                    LOGGER.error("Failed to process audit request '{}'", failedAuditString, ioException);
```

Comment:
Maybe also log about the timestampedFile? It might be we weren't able to write to this location

Reply:
I've added an extra error log about writing to a directory.